src/testrel.jl

```diff
@@ -633,17 +638,25 @@ function _test_rel_step(
         #   If an abort is expected it is encountered
         #   If no abort is expected it is not encountered
         #   If results are expected, they are found (other results are ignored)
-        #   If problems are expected, they are found (other problems are ignored)
-        #   If no problems are expected, warning level problems are ignored
+        #   If problems are expected, they are found
+        #   Unexpected problems with severity worse than allowable are not found
 
         unexpected_errors_found = false
+        error_levels = []
+        if step.allow_unexpected == :warning
+            push!(error_levels, "error")
+            push!(error_levels, "exception")
+        end
+        if step.allow_unexpected == :none
+            push!(error_levels, "warning")
+        end
+
         # Check if there were any unexpected errors/exceptions
         for problem in problems
             if contains_problem(step.expected_problems, problem)
                 @debug("$name: Expected problem", problem)
             else
-                unexpected_errors_found |= problem[:severity] == "error"
-                unexpected_errors_found |= problem[:severity] == "exception"
+                unexpected_errors_found |= in(problem[:severity], error_levels)
                 @info("$name: Unexpected problem: $(problem[:code])")
                 @debug("$name: Unexpected problem", problem)
```

Comment:
on line 670 it looks like if an problem is expected, then there can't be unexpected errors - is that still the desired behavior?

Reply:
Ah, no, and that would be why tests were passing.